Added mixpanel tracking for some basic events

fbarl opened the pull request:
Addresses #2460 by making use of Service UI's configured `window.mixpanel` variable directly.

##### Tracked events

* `scope.layout.refresh.click`
* `scope.layout.selector.click`
* `scope.layout.selector.keypress`
* `scope.metric.selector.pin.click`
* `scope.metric.selector.pin.next.keypress`
* `scope.metric.selector.pin.previous.keypress`
* `scope.metric.selector.unpin.click`
* `scope.metric.selector.unpin.keypress`
* `scope.node.click`
* `scope.node.relative.click`
* `scope.search.query.change`
* `scope.search.query.pin`
* `scope.topology.option.click`
* `scope.topology.selector.click`

Most of them have the active topology and view mode (layout) props attached to them to have some context of where the events are coming from. I checked it on the mixpanel dashboard and it seems to work nicely!

##### Other changes

* Replaced the `hitEnter` action with `pinSearch` which gets called directly from the `search.js`, since we only need to handle the *ENTER* key press from the search bar.
* Disable <kbd>r</kbd> shortcut when the resource view is disabled.

foot commented on client/app/scripts/components/search.js:
```diff
@@ -71,15 +75,36 @@ class Search extends React.Component {
+    // If the search query is parsable, pin it when ENTER key is hit.
+    if (ev.keyCode === ENTER_KEY_CODE && parseQuery(this.props.searchQuery)) {
+      trackMixpanelEvent('scope.search.query.pin', {
+        query: this.props.searchQuery,
```
User may be searching for service/process names which might be sensitive. While it would be interesting to know the type of query too (metrics vs property vs just a string), I don't think we should send the query itself.

fbarl replied:
Good point, I'll remove it!

foot commented on client/app/scripts/components/search.js:
```diff
@@ -71,15 +75,36 @@ class Search extends React.Component {
   doSearch(value) {
+    if (value !== '') {
+      trackMixpanelEvent('scope.search.query.change', {
+        query: value,
```
ditto, don't send user's info

fbarl replied:
Yes, I'll remove it here too!

fbarl commented:
@foot I addressed your comment and also added some tracking for `keypress` events (related to the already implemented `click` ones). Also, there was a tiny bug there that I fixed - pressing <kbd>r</kbd> when resource view was not available would still pin the first metric if none was pinned :)

All these changes were rather small so I'll merge this PR after the tests pass without your second look at it.
